### PR TITLE
MAINT: replace `progressbar` with `tqdm`

### PR DESCRIPTION
**Pull request by fkuehlein**
Deleted the `utils/progressbar` and everything that relates to it. Added `tqdm` dependency and replaced all `ProgressBar` instances with `tqdm` progress bars.

@ntfrgl, I'm not entirely sure I added the `tqdm` dependency in all the right places in `setup.cfg` and `.travis.yml`, could you please double check?

### CODE REVIEW

**Comment by fkuehlein** [comment]
CodeCov is complaining that overall test coverage is actually slightly reduced. That's because removing untested lines will not change the overall test coverage, while adding lines to untested methods – although way fewer – will reduce it. Of course I could add tests, but I'd say that's a different issue.

File: src/pyunicorn/climate/havlin.py [hunk 1]
```diff
@@ -172,19 +172,13 @@ def _calculate_correlation_strength(self, anomaly, max_delay, gamma=0.2):
...
         #  Calculate the inverse Fourier transform of all time series
         ifft = np.fft.ifft(anomaly, axis=0)
 
-        for i in range(N):
-            # Update progress bar every 10 steps
-            if self.silence_level <= 1:
-                if (i % 10) == 0:
-                    progress.update(i)
+        #  toggle progress bar
+        silence = self.silence_level > 1
```
Comment (ntfrgl) [inline src/pyunicorn/climate/havlin.py:179]: I would inline this into the `trange()` call.

Comment (ntfrgl) [reply src/pyunicorn/climate/havlin.py:179]: This obviously applies equally to the other changes.

**Comment by ntfrgl** [comment]
This coverage metric ended up slightly lower, because the coverage *within* the old `utils/progressbar`, which was entered via the removed lines in the *tested* methods, was somewhat higher than the current *overall average* over `pyunicorn`. Irrespectively of the particular metric, this PR is an improvement to the project code base.

For the record, the following are the *untested* methods in which the progress bar was replaced, and which appear not to be used anywhere else in the project. The last one in particular has been replaced by its Cythonized version, so I suggest deleting it.

- [`Network.GrowPreferentially‎()`](https://github.com/pik-copan/pyunicorn/blob/b04ef8ece443f452dca40edd02abab9df393edd9/src/pyunicorn/core/network.py#L1114)
- [`Network.GrowWeights‎()`](https://github.com/pik-copan/pyunicorn/blob/b04ef8ece443f452dca40edd02abab9df393edd9/src/pyunicorn/core/network.py#L1291)
- [`Surrogates‎.test_threshold_significance‎()`](https://github.com/pik-copan/pyunicorn/blob/b04ef8ece443f452dca40edd02abab9df393edd9/src/pyunicorn/timeseries/surrogates.py#L658)
- [`MutualInfoClimateNetwork._calculate_mutual_information‎()`](https://github.com/pik-copan/pyunicorn/blob/b04ef8ece443f452dca40edd02abab9df393edd9/src/pyunicorn/climate/mutual_info.py#L164)

I would also suggest to start writing the change log for the new release. This PR would be represented by a one-liner there.

Otherwise, this looks good to me.

**Comment by fkuehlein** [comment]
Thanks for the hints!

I improved the silencing of the new progressbars accordingly and removed `MutualInfoClimateNetwork._calculate_mutual_information‎()`. Made CodeCov slightly happier, too!

I wouldn't want to decide about the other apparently stale methods. Maybe @jdonges can have a quick look? Shouldn't keep us from merging this though.

**Comment by ntfrgl** [comment]
Thank you.

In contrast to `MutualInfoClimateNetwork._calculate_mutual_information‎()`, the other untested methods are public, so I don't see a reason for removing them at this point.

**Pull request merged**